app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt

```diff
@@ -33,11 +36,21 @@ class SkillDetailsPresenter(skillDetailsFragment: SkillDetailsFragment) : ISkill
     private var skillDetailsModel: SkillDetailsModel = SkillDetailsModel()
     private var skillDetailsView: ISkillDetailsView? = null
     private val utilModel: UtilModel = UtilModel(skillDetailsFragment.requireContext())
+    private var databaseRepository: IDatabaseRepository = DatabaseRepository()
+    private var settingView: ISettingsView? = null
 
     override fun onAttach(skillDetailsView: ISkillDetailsView) {
         this.skillDetailsView = skillDetailsView
     }
 
+    override fun loginLogout() {
```

Comment:
Duplicate. Extract the function in common module and use in both presenters

Reply:
@iamareebjamal can you please explain how to do this?

Reply:
B has a method

You want to call it in D

Create a component A which has that method, 

import A in B and in D